app/controllers/application_controller.rb

```diff
@@ -35,8 +35,17 @@ class MissingParameterError < Exception; end
 
   def set_language
     set_gettext_locale
-    unless LANGUAGES.include? FastGettext.locale
-      params[:locale] = FastGettext.default_locale
+    requested_locale = FastGettext.locale
+    unless LANGUAGES.include? requested_locale
+      # XXX: if the browser sent de_DE, try de
+      l = requested_locale.split('_')
+      if LANGUAGES.include?l[0]
+        params[:locale] = l[0]
+      else
+        # not sure we need this. Gettext handles missing locale
+        # just fine
+        params[:locale] = FastGettext.default_locale
+      end
       set_gettext_locale
```

Comment:
As it's written, I find the logic a little bit hard to follow. Would this be equivalent (I'm not sure)?
```ruby
def set_language
    # We only use the language, not the country part. So turn de_DE into de
    requested_locale = FastGettext.locale.split('_').first
    params[:locale] = requested_locale if LANGUAGES.include?(requested_locale)
    set_gettext_locale
    @lang = requested_locale
end
```

I hope it would work (just speculation, not tried at all) and is WAY more readable.